rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/policy/RestartDownNodesPolicy.java

```diff
@@ -47,73 +47,102 @@
     @Override
     public BooleanWrapper configure(Object... policyParameters) {
+
         BooleanWrapper parentConfigured = super.configure(policyParameters);
+
         if (policyParameters != null && policyParameters.length > 2) {
-            checkNodeStateEach = Integer.parseInt(policyParameters[2].toString());
+            this.checkNodeStateEach = Integer.parseInt(policyParameters[2].toString());
         }
+
+        this.timer = new Timer(TIMER_NAME);
+
         return parentConfigured;
     }
 
+    @Override
+    public void reconfigure(Object... policyParameters) {
+
+        this.timer.cancel();
+
+        this.configure(policyParameters);
+        this.scheduleRestartTimer();
+    }
+
     /**
      * Activates static policy. Register a listener in RMMonitoring
      */
     @Override
     public BooleanWrapper activate() {
+
         acquireAllNodes();
+        scheduleRestartTimer();
+
+        return new BooleanWrapper(true);
+    }
+
+    private void scheduleRestartTimer() {
+
+        this.timer.schedule(new TimerTask() {
 
-        timer.schedule(new TimerTask() {
             @Override
             public void run() {
-                int numberOfNodesToDeploy = 0;
-                for (Node downNode : nodeSource.getDownNodes()) {
-                    String nodeUrl = downNode.getNodeInformation().getURL();
-
-                    logger.info("Removing down node " + nodeUrl);
-                    BooleanWrapper removed = nodeSource.getRMCore().removeNode(nodeUrl, true);
-                    if (removed.getBooleanValue()) {
-                        logger.info("Down node removed " + nodeUrl);
-                        numberOfNodesToDeploy++;
-                    }
-                }
-
-                for (RMDeployingNode lostNode : nodeSource.getDeployingAndLostNodes()) {
-                    if (!lostNode.isLost()) {
-                        continue;
-                    }
-                    String nodeUrl = lostNode.getNodeURL();
-
-                    logger.info("Removing lost node " + nodeUrl);
-                    BooleanWrapper removed = nodeSource.getRMCore().removeNode(nodeUrl, true);
-                    if (removed.getBooleanValue()) {
-                        logger.info("Lost node removed " + nodeUrl);
-                        numberOfNodesToDeploy++;
-                    }
-                }
-
-                if (numberOfNodesToDeploy > 0) {
-                    logger.info("Acquiring " + numberOfNodesToDeploy + " nodes");
-                    acquireNodes(numberOfNodesToDeploy);
-                }
+                checkDownNodesAndReacquireNodesIfNeeded();
             }
-        }, checkNodeStateEach, checkNodeStateEach);
 
-        return new BooleanWrapper(true);
+        }, this.checkNodeStateEach, this.checkNodeStateEach);
+    }
+
+    private void checkDownNodesAndReacquireNodesIfNeeded() {
+
+        int numberOfNodesToDeploy = 0;
+
+        for (Node downNode : this.nodeSource.getDownNodes()) {
```

Comment:
can you clean up a bit this code? it smells like refactorable by removing duplications (not sure this word exists).